### PR TITLE
Fix PHPUnit deprecations

### PR DESCRIPTION
**Pull request by malarzm**
As per title :)

### CODE REVIEW

File: tests/Doctrine/Tests/Common/Proxy/ProxyLogicTest.php [hunk 1]
```diff
@@ -193,13 +192,16 @@ static function () use ($test) {
         self::assertSame('publicAssociationValue', $this->lazyObject->publicAssociation);
     }
 
+    /**
+     * @requires PHP > 7.1
```
Comment (malarzm) [inline tests/Doctrine/Tests/Common/Proxy/ProxyLogicTest.php:196]: `expectNotice` does not exist in PHPUnit running on 7.1. I figured the test can be skipped as this is tested in higher versions anyway but I can also check for `expectNotice` existence and conditionally call old method.

File: tests/Doctrine/Tests/Common/Proxy/ProxyLogicTest.php [hunk 2]
```diff
@@ -193,13 +192,16 @@ static function () use ($test) {
         self::assertSame('publicAssociationValue', $this->lazyObject->publicAssociation);
     }
 
+    /**
+     * @requires PHP >= 7.2
```
Comment (malarzm) [inline tests/Doctrine/Tests/Common/Proxy/ProxyLogicTest.php]: `expectNotice` does not exist in PHPUnit running on 7.1. I figured the test can be skipped as this is tested in higher versions anyway but I can also check for `expectNotice` existence and conditionally call old method.

Comment (derrabus) [reply tests/Doctrine/Tests/Common/Proxy/ProxyLogicTest.php]: > I can also check for `expectNotice` existence and conditionally call old method.

I'd prefer that. You could also polyfill `expectNotice` with a trait.

**Comment by derrabus** [comment]
For 3.3.x since it affects the test suite only?

**Review by derrabus — APPROVED** [review]

**Comment by derrabus** [comment]
Thanks!

**Pull request merged**